shared/util/codeql/util/Option.qll

```diff
@@ -0,0 +1,44 @@
+/** Provides a module for constructing optional versions of types. */
+
+/** A type with `toString`. */
+signature class TypeWithToString {
+  string toString();
+}
+
+/**
+ * Constructs an `Option` type that is a disjoint union of the given type and an
+ * additional singleton element.
+ */
+module Option<TypeWithToString T> {
```

Comment:
Should we also have an `OptionWithLocation` variant?

Reply:
Possibly? There are many use-cases without location, though, so I think we'll want the no-location version by default.

Reply:
Having locations for entities is crucial when debugging IMO.